app/battle/preparation.py

```diff
@@ -0,0 +1,23 @@
+def apply_armour(knight_data: dict) -> dict:
+    knight_data["protection"] = 0
+    if knight_data["armour"]:
+        for armour_dict in knight_data["armour"]:
+            knight_data["protection"] += armour_dict["protection"]
+
+    return knight_data
+
+
+def apply_weapon(knight_data: dict) -> dict:
+    knight_data["power"] += knight_data["weapon"]["power"]

```

Comment:
Ensure that 'weapon' and 'power' keys exist in 'knight_data' before accessing them to avoid potential KeyError exceptions.